pumps/mongo.go

```diff
@@ -115,159 +115,165 @@ func (m *MongoPump) connect() {
 	}
 
 	m.dbSession, err = mgo.DialWithInfo(dialInfo)
-	if err != nil {
+
+	// TODO - Should this not bail after a while?
+	for err != nil {
 		log.WithFields(logrus.Fields{
 			"prefix": mongoPrefix,
 		}).Error("Mongo connection failed:", err)
+
 		time.Sleep(5)
-		m.connect()
+		m.dbSession, err = mgo.DialWithInfo(dialInfo)
```

Comment:
Is there a situation where you would prefer the recursive approach to iterative?

Reply:
Quicksort & traversing trees.